pkg/metrics/statsd/cosmosdb/metrics.go

```diff
@@ -0,0 +1,90 @@
+package cosmodb
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+var _ http.RoundTripper = (*tracerRoundTripper)(nil)
+
+type tracerRoundTripper struct {
+	m   metrics.Interface
+	tr  http.RoundTripper
+	log *logrus.Entry
+}
+
+func New(log *logrus.Entry, tr *http.Transport, m metrics.Interface) *tracerRoundTripper {
+	return &tracerRoundTripper{
+		m:   m,
+		tr:  tr,
+		log: log,
+	}
+}
+
+func (t *tracerRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, err := t.tr.RoundTrip(r)
+	if resp == nil {
+		return resp, err
+	}
+
+	action := "unknown"
+	table := "unknown"
+	code := "unknown"
+	ru := float64(0)
+
+	url := r.URL.String()
+	// extract last path element: example.com/dbs/dbname/colls/OpenShiftClusters/{docs}
+	ld := strings.LastIndexByte(url, '/')
+	action = url[ld+1:]
+	// extract last path element from prev ref point: example.com/dbs/dbname/colls/{OpenShiftClusters}
+	table = url[strings.LastIndexByte(url[:ld], '/')+1:]
```

Comment:
this logic is still not making sense to me - I thought we decided to use the whole path, except that if the last item is an identifier, rewrite it to `{id}` or something to reduce cardinality?

Reply:
I think I misunderstood this. Whole path like `/dbs/{id}/colls/OpenShiftClusters/docs` ?